src/Umbraco.Community.CSPManager/Services/CspService.cs

```diff
@@ -4,80 +4,113 @@
-		IUmbracoContextAccessor umbracoContextAccessor, IScopeProvider scopeProvider)
+	public CspService(
+		IEventAggregator eventAggregator, 
+		IHostingEnvironment hostingEnvironment,
+		IUmbracoContextAccessor umbracoContextAccessor, 
+		IScopeProvider scopeProvider,
+		AppCaches appCaches)
 	{
 		_eventAggregator = eventAggregator;
 		_hostingEnvironment = hostingEnvironment;
 		_umbracoContextAccessor = umbracoContextAccessor;
 		_scopeProvider = scopeProvider;
+		_runtimeCache = appCaches.RuntimeCache;
 	}
 
-	public CspDefinition? GetCspDefinition(HttpContext httpContext)
-	{
-		CspDefinition? definition = null;
+	public async Task<CspDefinition?> GetCspDefinitionAsync(bool isBackOfficeRequest)
+	{		
 		using var scope = _scopeProvider.CreateScope();
-		if (_umbracoContextAccessor.TryGetUmbracoContext(out var umbracoContext))
-		{
-			if (umbracoContext.IsFrontEndUmbracoRequest())
-			{
-				definition = GetDefinition(scope, false) ??
-				             new CspDefinition { Enabled = false, IsBackOffice = false };
-			}
-		}
-
-		if (httpContext.Request.IsBackOfficeRequest())
-		{
-			//TODO: Oembed providers - https://our.umbraco.com/documentation/extending/Embedded-Media-Provider/
-			definition = GetDefinition(scope, true) ?? new CspDefinition { Enabled = false, IsBackOffice = true };
-		}
-
+		
+		//TODO: Oembed providers - https://our.umbraco.com/documentation/extending/Embedded-Media-Provider/
+		CspDefinition? definition = await GetDefinitionAsync(scope, isBackOfficeRequest)
+			?? new CspDefinition { 
+				Id = isBackOfficeRequest ? CspConstants.DefaultBackofficeId : CspConstants.DefaultFrontEndId,
+				Enabled = false,
+				IsBackOffice = isBackOfficeRequest 
+			};
+			
 		AddWebsocketsForAspNet(definition);
 
-		_eventAggregator.Publish(new CspWritingNotification(definition, httpContext));
-
 		scope.Complete();
 		return definition;
 	}
-	
-	private static CspDefinition? GetDefinition(IScope scope, bool isBackOffice)
+
+	public async Task<CspDefinition?> GetCachedCspDefinitionAsync(bool isBackOfficeRequest)
+	{
+		string cacheKey = isBackOfficeRequest ? CspConstants.BackOfficeCacheKey : CspConstants.FrontEndCacheKey;
+
+		return await _runtimeCache.GetCacheItem(cacheKey, async () =>
+		{
+			return await GetCspDefinitionAsync(isBackOfficeRequest);
+		});
+	}
+
+	private static async Task<CspDefinition?> GetDefinitionAsync(IScope scope, bool isBackOffice)
 	{
 		var sql = scope.SqlContext.Sql()
 			.SelectAll()
 			.From<CspDefinition>()
 			.LeftJoin<CspDefinitionSource>()
 			.On<CspDefinition, CspDefinitionSource>((d, s) => d.Id == s.DefinitionId)
-			.Where<CspDefinition>(x => x.IsBackOffice == isBackOffice && x.Enabled == true);
+			.Where<CspDefinition>(x => x.IsBackOffice == isBackOffice);
+
 		var raw = sql.SQL;
-		var data = scope.Database.FetchOneToMany<CspDefinition>(c => c.Sources, sql);
+		var data = await Task.FromResult(scope.Database.FetchOneToMany<CspDefinition>(c => c.Sources, sql));
```

Comment:
No need to force async/await on this method it can just return Task<CspDefinition?> by movingthe Task.FromResult to be around the FirstOrDefault() call.